File: deb/openmediavault-transmissionbt/var/www/openmediavault/js/omv/module/transmissionbt/manage/dialog/delete.js
/**
 * This file is part of OpenMediaVault.
 *
 * @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
 * @author    Marcel Beck <marcel.beck@mbeck.org>
 * @copyright Copyright (c) 2011-2012 Marcel Beck
 *
 * OpenMediaVault is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * any later version.
 *
 * OpenMediaVault is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with OpenMediaVault. If not, see <http://www.gnu.org/licenses/>.
 */
// require("js/omv/Window.js")

Ext.ns("OMV.TransmissionBT");

/**
 * @class OMV.TransmissionBT.DeleteDialog
 * @config title The dialog title.
 * @config service The name of the RPC service.
 * @config method The name of the RPC method.
 * @config params Additional RPC method parameters.
 * @config waitMsg The displayed waiting message.
 */
OMV.TransmissionBT.DeleteDialog = function(config) {
	var initialConfig = {
		url: this.url,
		title: "Delete torrent",
		waitMsg: "Deleting torrent ...",
		width: 450,
		autoHeight: true,
		layout: "fit",
		modal: true,
		border: false,
		buttonAlign: "center",
		buttons: [{
			text: "OK",
			handler: this.cbOkBtnHdl.createDelegate(this),
			scope: this
		},{
			text: "Cancel",
			handler: this.cbCancelBtnHdl.createDelegate(this),
			scope: this
		}]
	};
	Ext.apply(initialConfig, config);
	OMV.TransmissionBT.DeleteDialog.superclass.constructor.call(this, initialConfig);
	this.addEvents(
		/**
		 * Fires after the installation has been finished successful.
		 */
		"success",
		"before"
	);
};
Ext.extend(OMV.TransmissionBT.DeleteDialog, OMV.Window, {
	initComponent : function() {
		this.form = new Ext.form.FormPanel({
			frame: true,
			border: false,
			layout: "form",
			defaults: {
				anchor: "100%",
				labelSeparator: ""
			},
			autoHeight: true,
			fileUpload: true,
			items: [{
				xtype: "hidden",
				name: "service",
				value: this.service
			},{
				xtype: "hidden",
				name: "method",
				value: this.method
			},{
				xtype: "hidden",
				name: "params",
				value: this.params
			},{
				xtype: "checkbox",
				name: "delete-local-data",
				id: "delete-local-data",
				fieldLabel: "Delete Local Data",
				checked: false,
				inputValue: 1
			}]
		});
		this.items = this.form;
		OMV.TransmissionBT.DeleteDialog.superclass.initComponent.apply(this, arguments);
	},

	/**
	 * @method cbOkBtnHdl
	 * Method that is called when the 'OK' button is pressed.
	 */
	cbOkBtnHdl : function() {
		var basicForm = this.form.getForm();
		if (!basicForm.isValid()) {
			return;
		}
		var delete_local_data = Ext.getCmp('delete-local-data').getValue();

		this.fireEvent("success", this, delete_local_data);
		this.close();
	},

	/**
	 * @method cbCancelBtnHdl
	 * Method that is called when the 'Cancel' button is pressed.
	 */
	cbCancelBtnHdl : function() {
		this.close();
	}
});

File: deb/openmediavault/usr/share/openmediavault/confdbadm/commands.d/delete.py
#!/usr/bin/env python3
#
# This file is part of OpenMediaVault.
#
# @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
# @author    Volker Theile <volker.theile@openmediavault.org>
# @copyright Copyright (c) 2009-2025 Volker Theile
#
# OpenMediaVault is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# any later version.
#
# OpenMediaVault is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with OpenMediaVault. If not, see <http://www.gnu.org/licenses/>.
import argparse
import os.path
import sys

import openmediavault.log

import openmediavault


class Command(
    openmediavault.confdbadm.ICommand, openmediavault.confdbadm.CommandHelper
):
    @property
    def description(self):
        return "Delete a configuration object."

    def execute(self, *args):
        rc = 0
        # Parse the command line arguments.
        parser = argparse.ArgumentParser(
            prog="%s %s" % (os.path.basename(args[0]), args[1]),
            description=self.description,
        )
        parser.add_argument(
            "id",
            type=self.argparse_is_datamodel_id,
            help="The data model ID, e.g. 'conf.service.ssh'",
        )
        group = parser.add_mutually_exclusive_group()
        group.add_argument("--uuid", nargs="?", type=self.argparse_is_uuid4)
        group.add_argument("--filter", nargs="?", type=self.argparse_is_json)
        cmd_args = parser.parse_args(args[2:])
        # Create a backup of the configuration database.
        self.create_backup()
        # Get the database.
        db = openmediavault.config.Database()
        try:
            if cmd_args.filter:
                # Create the query filter.
                db_filter = openmediavault.config.DatabaseFilter(
                    cmd_args.filter
                )
                objs = db.delete_by_filter(cmd_args.id, db_filter)
            else:
                # Query the database.
                objs = db.get(cmd_args.id, cmd_args.uuid)
                if not isinstance(objs, list):
                    if objs is None:
                        objs = []
                    else:
                        objs = [objs]
                # Delete the configuration object(s).
                for obj in objs:
                    db.delete(obj)
        except Exception as e:
            rc = 1
            # Display the exception message.
            openmediavault.log.error(
                "Failed to delete the configuration object: %s", str(e)
            )
            # Rollback all changes.
            self.rollback_changes()
        finally:
            # Unlink the configuration database backup.
            self.unlink_backup()
            # Output the number of deleted configuration objects.
            if 0 == rc:
                print("Deleted %d object(s)" % len(objs))
        return rc


if __name__ == "__main__":
    sys.argv.insert(1, "delete")
    command = Command()
    rc = command.execute(*sys.argv)
    sys.exit(rc)
